concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/30 Save, Cancel, and Custom Buttons.md

```diff
@@ -0,0 +1,23 @@
+Form and popup editing modes include built-in Save and Cancel buttons. If you want to execute custom code on click one of these buttons, specify the [onSaving](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onSaving) or [onEditCanceling](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditCanceling) event handler correspondingly. Assign `true` to the **e.cancel** parameter to cancel the Save or Cancel operations and keep the form or popup opened.
+
+Specify the **editing.texts**.[cancelRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#cancelRowChanges) and **editing.texts**.[saveRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#saveRowChanges) properties to change texts of the Save and Cancel buttons.
+
+To customize the Save and Cancel buttons, re-define them:
+
+- Popup edit mode     
+Add the **editing**.[popup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#popup) object to the DataGrid configuration and declare the Save, Cancel and custom buttons in the **popup**.[toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) array.
+
+- Form edit mode     
+Add the **editing**.[form](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#form) object to the DataGrid configuration and use the **form.items** array to arrange all items you want to display in the edit form. Then, add a [simple item](/Documentation/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/) with a template containing a standalone [Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/) component. Declare the buttons in this toolbar. Use the following CSS to hide the built-in Save and Cancel buttons:
+
+        #gridForm .dx-datagrid-form-buttons-container {  
+            display: none  
+        }
+
+Finally, handle the button onClick event to call the [saveEditData](/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#saveEditData) and [cancelEditData](/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#cancelEditData) methods correspondingly. 
```

Comment:
Please do not say "correspondingly". Say "respectively". Thanks in advance.

Note that "respectively" is a separate clause in a sentence, therefore, it requires a comma.

> **the** `saveEditData` and `cancelEditData` methods

The same incorrect article case.

```suggestion
Finally, handle the button onClick event to call [saveEditData](/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#saveEditData) and [cancelEditData](/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#cancelEditData) methods, respectively. 
```